### PR TITLE
Add libmicrohttpd test

### PR DESCRIPTION
**Pull request by mitiao**
verified result:
http://147.2.207.208/tests/163

### CODE REVIEW

File: tests/console/libmicrohttpd.pm [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    clear_console;
+    assert_script_run "mkdir -p /etc/openvas/cert";
+    assert_script_run "cd /etc/openvas/cert";
+    assert_script_run "echo -e \"\n\n\n\n\n\n\" | openssl req -new -x509 -newkey rsa:2048 -keyout gsa.key -days 3560 -out gsa.cert -nodes";
+    assert_script_run "ls gsa.key gsa.cert";
+
+    # Start greenbone-security-assistant
+    clear_console;
+    assert_script_run "/usr/sbin/gsad --listen=127.0.0.1 --port=9392 --alisten=127.0.0.1 --aport=9393 --mlisten=127.0.0.1 --mport=9390 --ssl-private-key=/etc/openvas/cert/gsa.key --ssl-certificate=/etc/openvas/cert/gsa.cert &> gsad.log";
+    assert_script_run "cat gsad.log | tee /dev/$serialdev";
```
Comment (coolo) [inline tests/console/libmicrohttpd.pm]: why not upload_logs instead of this?

File: tests/console/libmicrohttpd.pm [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    # Create slef-signed certificates
+    clear_console;
+    assert_script_run "mkdir -p /etc/openvas/cert";
+    assert_script_run "cd /etc/openvas/cert";
+    assert_script_run "echo -e \"\n\n\n\n\n\n\" | openssl req -new -x509 -newkey rsa:2048 -keyout gsa.key -days 3560 -out gsa.cert -nodes";
+    assert_script_run "ls gsa.key gsa.cert";
+
+    # Start greenbone-security-assistant
+    clear_console;
+    assert_script_run "/usr/sbin/gsad --listen=127.0.0.1 --port=9392 --alisten=127.0.0.1 --aport=9393 --mlisten=127.0.0.1 --mport=9390 --ssl-private-key=/etc/openvas/cert/gsa.key --ssl-certificate=/etc/openvas/cert/gsa.cert &> gsad.log";
```
Comment (coolo) [inline tests/console/libmicrohttpd.pm]: I think you should tear it down afterwards.

File: tests/console/libmicrohttpd.pm (hunk 1 above)
Comment (mitiao) [inline tests/console/libmicrohttpd.pm]: done.

File: tests/console/libmicrohttpd.pm (hunk 2 above)
Comment (mitiao) [inline tests/console/libmicrohttpd.pm]: probably :)
so far this is the only way to start the service

File: tests/console/libmicrohttpd.pm [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+use testapi;
+use utils;
+
+sub run() {
+    select_console 'root-console';
+
+    # Install greenbone-security-assistant
+    zypper_call("in greenbone-security-assistant");
+
+    # Create slef-signed certificates
```
Comment (okurz) [inline tests/console/libmicrohttpd.pm]: s/slef/self/

Comment (mitiao) [inline tests/console/libmicrohttpd.pm]: fixed

File: tests/console/libmicrohttpd.pm (hunk 2 above)
Comment (mitiao) [inline tests/console/libmicrohttpd.pm]: ah, i see.
done.

File: tests/console/libmicrohttpd.pm [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    # Start greenbone-security-assistant
+    clear_console;
+    assert_script_run "/usr/sbin/gsad --listen=127.0.0.1 --port=9392 --alisten=127.0.0.1 --aport=9393 --mlisten=127.0.0.1 --mport=9390 --ssl-private-key=/etc/openvas/cert/gsa.key --ssl-certificate=/etc/openvas/cert/gsa.cert &> gsad.log";
+    upload_logs "/etc/openvas/cert/gsad.log";
+
+    # Check login page of Greenbone Security Assistant Web
+    validate_script_output "curl -k https://localhost:9392/login/login.html", sub { m/Greenbone Security Assistant/ };
+
+    # Stop greenbone-security-assistant
+    assert_script_run "killall -9 gsad";
```
Comment (okurz) [inline tests/console/libmicrohttpd.pm]: I am not sure why you use '-9'. I guess just TERM should suffice but ok for me.

Comment (mitiao) [inline tests/console/libmicrohttpd.pm]: remove -9 since it's sufficient.

**Pull request merged**